flang/lib/Lower/OpenMP/ClauseProcessor.cpp

```diff
@@ -999,6 +1000,69 @@ bool ClauseProcessor::processUseDevicePtr(
       });
 }
 
+static llvm::StringMap<bool> getTargetFeatures(mlir::ModuleOp module) {
+  llvm::StringMap<bool> featuresMap;
+  llvm::SmallVector<llvm::StringRef> targetFeaturesVec;
+  if (mlir::LLVM::TargetFeaturesAttr features =
+          fir::getTargetFeatures(module)) {
+    llvm::StringRef targetFeaturesStr(features.getFeaturesString());
```

Comment:
Wouldn't it help to call `features.getFeatures()` here instead and iterate through the list of `StringAttr` directly, rather than producing the comma-separated string to then split it again?

Reply:
Yeah, that seems cleaner. Updated code.